conda/cli/install.py

```diff
@@ -56,6 +56,10 @@ def check_prefix(prefix, json=False):
         raise CondaValueError(
             f"Cannot create a conda environment with '{os.pathsep}' in the prefix. Aborting."
         )
+    if prefix.endswith("/envs"):
```

Comment:
I'm hesitant of making the condition this broad, we don't care if `/some/other/path/envs` is used or even `conda create --name envs`, we only care if `$CONDA_PREFIX/envs` is used.

I believe a better option is to check if `prefix.parent` is a conda env and if so to disallow certain protected directory names (`bin`, `etc`, `conda-meta`, `envs`, etc.).

We also should account for a symlink pointing to `$CONDA_PREFIX/envs`.

Reply:
cc @chenghlee